src/core/stdcpp/string.d

```diff
@@ -167,6 +167,24 @@ extern(D):
     ///
     inout(T)[] opIndex() inout pure nothrow @safe @nogc                     { return as_array(); }
 
+    /// Two `basic_string`s are equal if they represent the same sequence of code units.
+    bool opEquals(scope const ref basic_string s) const pure nothrow @safe  { return as_array == s.as_array; }
+    /// ditto
+    bool opEquals(basic_string s) const                                     { return as_array == rhs.as_array; }
+    /// ditto
+    bool opEquals(scope const T[] s) const pure nothrow @safe               { return as_array == s; }
+
+    /// Performs lexicographical comparison.
+    bool opCmp(scope const ref basic_string rhs) const pure nothrow @safe   { return __cmp(as_array, rhs.as_array); }
+    /// ditto
+    bool opCmp(basic_string rhs) const                                      { return __cmp(as_array, rhs.as_array); }
+    /// ditto
+    bool opCmp(scope const T[] rhs) const pure nothrow @safe                { return __cmp(as_array, rhs); }
+
+    /// Hash to allow `basic_string`s to be used as keys for built-in associative arrays.
+    /// **The result will generally not be the same as C++ `std::hash<std::basic_string<T>>`.**
```

Comment:
Maybe we should work this so that the result is the same as `std::hash`, although this is good for now.